src/Form/Fields/Formatters/UploadFormatter.php

```diff
@@ -18,8 +19,20 @@ public function setAlwaysReturnFullObject(?bool $returnFullObject = true): self
         return $this;
     }
 
+    /**
+     * @param  IsUploadField  $field
+     */
     public function toFront(SharpFormField $field, $value)
     {
+        // in editor embeds case, we add some extra data to handle the file afterwards
+        if (is_array($value) && $this->alwaysReturnFullObject) {
+            return [
+                ...$value,
+                'shouldOptimizeImage' => $field->isShouldOptimizeImage(),
+                'transformOriginal' => $field->isTransformOriginal(),
+            ];
```

Comment:
@dvlpp For editor uploads / embeds, we add those attributes to be available in FormController to dispatch jobs